src/store-tree/index.ls

```diff
@@ -40,45 +37,34 @@ class StoreTree extends StoreNode
 
 
   $is-loading: ->
-    for own key of @_children when @[key].$is-loading!
-      return yes
-    no
+    @_children |> values |> any (.$is-loading!)
 
 
   $reset: !-> @$batch-emit-updates ~>
-    @_for-each-subnode (subnode) -> subnode.$reset!
+    @_children |> Obj.each (.$reset!)
 
 
   $set: (data) !-> @$batch-emit-updates ~>
     switch typeof! data
-    | \Object   => @_for-each-subnode (subnode, key) -> subnode.$set data[key] if data[key] isnt undefined
-    | \Null     => @_for-each-subnode (subnode) -> subnode.$set null
+    | \Object   => @_children |> obj-to-pairs |> each ([key, subnode]) -> subnode.$set data[key] if data[key] isnt undefined
```

Comment:
The `data[key] isnt undefined` will still be needed. Otherwise yes that looks good. I'm going to break this out into a function so this doesn't need to be a single line